akka-javasdk-tests/src/test/java/akkajavasdk/components/agent/AgentIntegrationTest.java

```diff
@@ -86,6 +87,50 @@ public void shouldMapFailedJsonParsingResponse() {
     assertThat(result.response()).isEqualTo("default response");
   }
 
+
+  @Test
+  public void shouldCallASingleTool() {
+
+    var userQuestion = "How is the weather today in Leuven?";
+
+    //when asking for the weather, it first look up for today's date
+    testModelProvider.mockResponse(
+      s -> s.equals(userQuestion),
+      new TestModelProvider.AiResponse(
+        new TestModelProvider.ToolInvocationRequest("getDateOfToday", "")));
+
+    // when receiving the date back, model asks for calling getWeather
+    testModelProvider.mockResponseToToolResult(
```

Comment:
maybe we could make it more user-friendly with a fluent builder pattern:
```
testModelProvider.when(result -> result.content().equals("2025-01-01"))
                               .thenReplyWithToolRequest(...)
```
and to cover to one below:
```
testModelProvider.when(result -> result.content().startsWith("The weather is"))
                               .thenReplyWithResult()
```

Reply:
BTW we shouldn't solve that in that PR.